tensor_genn/algorithms/relu_ann.py

```diff
@@ -242,31 +246,62 @@ def convert(self, tf_model, scaled_tf_weights=None):
 
         return self.g_model, self.neuron_pops, self.current_source
 
-    def evaluate(self, X, y=None):
+    def evaluate(self, X, y=None, raster_plot=False):
         n_examples = len(X)
         X = X.reshape(n_examples,-1)
         y = y.reshape(n_examples)
         n = len(self.neuron_pops)
         
         n_correct = 0
+        spike_ids = [None]*len(self.neuron_pops)
+        spike_times = [None]*len(self.neuron_pops)
         for i in range(n_examples):
             # Before simulation
             for j, npop in enumerate(self.neuron_pops):
-                    npop.vars["SpikeNumber"].view[:] = 0
-                    npop.vars["Vmem"].view[:] = random.uniform(self.Vres,self.Vthr)
-                    self.g_model.push_state_to_device("if"+str(j))
+                npop.vars["SpikeNumber"].view[:] = 0
+                npop.vars["Vmem"].view[:] = random.uniform(self.Vres,self.Vthr)
+                self.g_model.push_state_to_device("if"+str(j))
                 
             self.current_source.vars['magnitude'].view[:] = X[i]*(-self.Vthr * (self.dCm/self.timestep))
             self.g_model.push_var_to_device("cs",'magnitude')
 
             # Run simulation
-            for _ in range(math.ceil(self.single_example_time/self.timestep)):
+            for t in range(math.ceil(self.single_example_time/self.timestep)):
                 self.g_model.step_time()
 
+                if raster_plot and i == n_examples-1:
+                    for j,npop in enumerate(self.neuron_pops):
+                        self.g_model.pull_current_spikes_from_device(npop.name)
+
+                        ids = npop.current_spikes # size of npop
+                        ts = np.ones(ids.shape) * t # size of npop
+                    
+                        if spike_ids[j] is None:
+                            spike_ids[j] = np.copy(ids)
+                            spike_times[j] = ts
+                        else:
+                            spike_ids[j] = np.hstack((spike_ids[j],ids))
+                            spike_times[j] = np.hstack((spike_times[j], ts))
+
             # After simulation
             self.g_model.pull_var_from_device("if"+str(n-1),'SpikeNumber')
             SpikeNumber_view = self.neuron_pops[-1].vars["SpikeNumber"].view
             n_correct += (np.argmax(SpikeNumber_view)==y[i])
+
         accuracy = (n_correct / n_examples) * 100.
+
+        if raster_plot:
```

Comment:
You could then make a ``tensor_genn.utils.plotting`` module and have the functionality to plot your spike data structure in there.